### ISSUE TITLE
HA Automation calls to service.toggle_cover fail

### ISSUE THREAD

**Issue by andyrak** [andyrak/hacs-aladdin-legacy#26]
## System Information

version | core-2024.4.4
-- | --
installation_type | Home Assistant Container
dev | false
hassio | false
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.1.21-v8+
arch | aarch64
timezone | America/Los_Angeles
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4973
Installed Version | 1.34.0
Stage | running
Available Repositories | 1408
Downloaded Repositories | 35

</details>

<details><summary>AccuWeather</summary>

can_reach_server | ok
-- | --
remaining_requests | 29

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | August 8, 2024 at 17:00
relayer_connected | true
relayer_region | us-east-1
remote_enabled | true
remote_connected | true
alexa_enabled | true
google_enabled | false
remote_server | us-east-1-8.ui.nabu.casa
certificate_status | ready
instance_id | 20738baa02714c0f95f64db3bc7b910f
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Dashboards</summary>

dashboards | 13
-- | --
resources | 14
views | 14
mode | yaml

</details>

<details><summary>NextDNS</summary>

can_reach_server | ok
-- | --

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | April 23, 2024 at 03:35
-- | --
current_recorder_run | April 29, 2024 at 23:45
estimated_db_size | 2822.11 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/andyrak/hacs-aladdin-legacy/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

When calling the `service.cover_toggle`, the aladdin connect plugin does not correctly detect the current state, and fails with a bad request.

### Reproduction steps

1. Create automation with service.toggle_cover
```
service: cover.toggle
target:
  device_id: <redacted_device_id>
data: {}
```
2. Call automation
3. Encounter errors as listed in debug logs section

### Debug logs

```text
This error originated from a custom integration.

Logger: custom_components.aladdin_connect
Source: custom_components/aladdin_connect/api.py:220
integration: Aladdin Connect Legacy (documentation, issues)
First occurred: April 30, 2024 at 16:05:07 (6 occurrences)
Last logged: 11:34:05

[API] An error occurred sending command OPEN_DOOR to door Right Garage Door; 400, message='Bad Request', url=URL('https://api.smartgarage.systems/command/devices/F0AD4E264D9D/doors/1')
[API] An error occurred sending command OPEN_DOOR to door Left Garage Door; 400, message='Bad Request', url=URL('https://api.smartgarage.systems/command/devices/F0AD4E2649F4/doors/1')

---------

This error originated from a custom integration.

Logger: custom_components.aladdin_connect
Source: custom_components/aladdin_connect/api.py:213
integration: Aladdin Connect Legacy (documentation, issues)
First occurred: April 30, 2024 at 16:05:07 (6 occurrences)
Last logged: 11:34:05

[API] An error occurred sending command OPEN_DOOR to door Right Garage Door; {"error":"Door is already open","type":"BusinessLogicError"}
[API] An error occurred sending command OPEN_DOOR to door Left Garage Door; {"error":"Door is already open","type":"BusinessLogicError"}
```

**Comment by w9nl** [comment]
I'm seeing the same issue.  I have a automation where my door will open when I (my phone) enters the zone created for my home.  When the door is then manually closed it will still show as opened in HA.  If I hit close in HA I get a message Failed to call service cover/close_cover. Aladdin Connect API failed to close the cover.  If I hit open in HA the door does open and then I can close in HA via a close command button.  Not sure if this helps any or not.

**Comment by andyrak** [comment]
Thanks for adding a report, @w9nl. I've been busy the last couple of weeks, so I'm hoping to pick this up this week.

**Comment by andyrak** [comment]
Fixed in v0.2.2.
